### PR TITLE
188 compare results with and without user changes

### PR DESCRIPTION
**Pull request by markushal**
@joAschauer please have a look at this. I implemented a way to compare costs with and without considering user data changes. I find this very helpful in understanding the effects of modifying input data. 

I am sure the implementation and representation of the data can still be improved...

### CODE REVIEW

File: app/layout_elements.py [hunk 1]
```diff
@@ -20,36 +21,55 @@ def display_costs(
...
-    )
+    # select which dataset to display:
+    if st.session_state["user_changes_df"] is not None:
+        with c2:
+            st.info("Input data has been modified. Select which data to display.")
+            select_data = st.radio(
+                "Select data to display",
+                ["with user modifications", "without user modifications", "difference"],
+                horizontal=True,
+                key=f"select_data_{key}_{key_suffix}",
```
Comment (joAschauer) [inline app/layout_elements.py:34]: you get duplicate key error when clicking manual select in the left column 
```suggestion
                key=f"select_user_modificatons_data_{key}_{key_suffix}",

```

File: app/layout_elements.py [hunk 2]
```diff
@@ -20,36 +21,55 @@ def display_costs(
...
-    show_which_data = st.radio(
-        "Select elements to display:",
-        ["All", "Manual select"],
-        index=0,
-        horizontal=True,
-        key=f"show_which_data_{key}_{key_suffix}",
-    )
+    # select which dataset to display:
+    if st.session_state["user_changes_df"] is not None:
+        with c2:
```
Comment (joAschauer) [inline app/layout_elements.py:28]: I would prefer to put the user modification selection below and and not side by side

File: app/layout_elements.py [hunk 3]
```diff
@@ -20,36 +21,55 @@ def display_costs(
...
+            st.info("Input data has been modified. Select which data to display.")
+            select_data = st.radio(
+                "Select data to display",
+                ["with user modifications", "without user modifications", "difference"],
```
Comment (joAschauer) [inline app/layout_elements.py:32]:
```suggestion
                ["With Modifications", "Without Modifications", "Difference"],

```

File: app/layout_elements.py [hunk 4]
```diff
@@ -20,36 +21,55 @@ def display_costs(
...
+            df_res = df_res.loc[ind_select]
+
+        # sort:
+        sort_ascending = st.toggle(
+            "Sort by total costs?",
+            value=True,
+            key=f"sort_data_{key}_{key_suffix}",
+        )
+        if sort_ascending:
+            df_res = df_res.sort_values(["Total"], ascending=True)
```
Comment (joAschauer) [inline app/layout_elements.py:72]: can be moved outside c1
```suggestion
    if sort_ascending:
        df_res = df_res.sort_values(["Total"], ascending=True)

```

**Review by joAschauer — CHANGES_REQUESTED** [review]

File: app/layout_elements.py (hunk 2 above)
Comment (markushal) [reply app/layout_elements.py:28]: I would prefer to leave it like it is for now

**Comment by markushal** [comment]
@joAschauer I applied your suggestions but kept the column layout like it is. For now.

**Review by joAschauer — APPROVED** [review]

**Pull request merged**